### DISCUSSION TITLE
Microservices:PersonService is executed everyday at midnight and can't be disabled

### DISCUSSION THREAD

**Discussion by ilopmar** [immich-app/immich#11078] [Q&A]
### The bug

I see in the logs that everyday at midnight there is an schedule job to find find people on photos.

```
immich_server  | [Nest] 8  - 07/13/2024, 12:00:00 AM   DEBUG [Microservices:PersonService] Deleted 0 people
immich_server  | [Nest] 8  - 07/13/2024, 12:00:08 AM   DEBUG [Microservices:PersonService] Face bea9abdf-dc3f-4bd1-9fd3-334e42c889d7 only matched the face itself, skipping
immich_server  | [Nest] 8  - 07/13/2024, 12:00:08 AM   DEBUG [Microservices:PersonService] Face ab9b5328-fa1b-41e0-b351-769f696c4ac9 only matched the face itself, skipping
immich_server  | [Nest] 8  - 07/13/2024, 12:00:08 AM   DEBUG [Microservices:PersonService] Face ff9aaa14-7fad-4ccd-865b-2e813aaa7ae5 only matched the face itself, skipping
immich_server  | [Nest] 8  - 07/13/2024, 12:00:08 AM   DEBUG [Microservices:PersonService] Face 80542732-0c3e-46b8-aa77-b48aba0b6bd5 only matched the face itself, skipping
immich_server  | [Nest] 8  - 07/13/2024, 12:00:08 AM   DEBUG [Microservices:PersonService] Face 81ac0327-cb76-4c27-8167-cbbcfa349d45 only matched the face itself, skipping
....
....
```

In my case the process takes over 30 minutes to finish because I have over 100K pictures, all in different external libraries. This puts my home-lab server into high CPU usage for all that time. That is executed even if I haven't uploaded new pictures, so it's a waste of resources.
In the Admin UI I have disabled the cron to scan external libraries, but I can't find a way to disable that other cron (or change it's execution).

What's the rationale of doing this every night without being able to disable it?

### The OS that Immich Server is running on

Ubunbu Server 22.04.4

### Version of Immich Server

v1.107.2

### Version of Immich Mobile App

v1.107

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
NA
```

### Your .env content

```Shell
NA
```

### Reproduction steps

```bash
1. Upload some pictures
2. Wait until midnight. The job to look for people is executed
3. Next day is executed again even if nothing changed
```

**Comment by mertalev** [comment]
This step isn't optional: it's the other half of the facial recognition algorithm. That being said, you're right that it doesn't need to run if there are no new assets. Specifically, it only needs to run when a new face has been added since its last run.

**Reply by ilopmar** [reply]
I've seend this when refreshing an external library. I think it's because first the facial recognition is run and then all the result are processed/merged (or something similar). For me specifically I don't really need this second step because when I add new files to the external libraries directories I send a request using the API to refresh that library, so that step is executed. 
If this can't be disabled, at least, as you mention it should be smart enough to run only when needed.

P.S: I'm tempted to create a crontab to stop Immich everyday at 23:59 and start it again at 00:01 to avoid this.

**Reply by mertalev** [reply]
This should be improved with #11080.

**Reply by ilopmar** [reply]
Awesome! :tada:
